src/test/java/com/lithium/dbi/rdbi/recipes/scheduler/DedupJobSchedulerTest.java

```diff
@@ -0,0 +1,318 @@
+        assertEquals(result1.getJobStr(), "{hello:world}");
+
+        // Delete the job thats in the running queue.
+        assertTrue(scheduledJobSystem.deleteJob(tubeName, result1.getJobStr()));
+
+        // Verify you cant reserve any jobs.
+        JobInfo result2 = scheduledJobSystem.reserveSingle(tubeName, 1000);
+        assertNull(result2);
+
+        // Schedule the job again.
+        scheduledJobSystem.schedule(tubeName, "{hello:world}", 0);
+
+        //Delete job while its in the ready queue.
+        assertTrue(scheduledJobSystem.deleteJob(tubeName, "{hello:world}"));
+
+        // Try to reserve and shouldn't be able to.
+        JobInfo result4 = scheduledJobSystem.reserveSingle(tubeName, 1000);
+        assertNull(result4);
+    }
+
+    @Test
+    public void testAck() throws InterruptedException {
+        // Schedule.
+        assertTrue(scheduledJobSystem.schedule(tubeName, "{hello:world}", 0));
+
+        // Verify the job is in the ready queue.
+        assertTrue(scheduledJobSystem.inReadyQueue(tubeName, "{hello:world}"));
+        assertFalse(scheduledJobSystem.inRunningQueue(tubeName, "{hello:world}"));
+
+        // Attempt to ack non running job. Should fail
+        assertFalse(scheduledJobSystem.ackJob(tubeName, "{hello:world}"));
+
+        JobInfo result1 = scheduledJobSystem.reserveSingle(tubeName, 1000);
+        assertEquals(result1.getJobStr(), "{hello:world}");
+
+        // Verify the job is no longer in the ready queue.
+        assertFalse(scheduledJobSystem.inReadyQueue(tubeName, "{hello:world}"));
+        assertTrue(scheduledJobSystem.inRunningQueue(tubeName, "{hello:world}"));
+
+        // Schedule the job again. Since its now reserved should be able to schedule it again.
+        assertTrue(scheduledJobSystem.schedule(tubeName, "{hello:world}", 0));
+
+        // Verify the job is in the ready queue now.
+        assertTrue(scheduledJobSystem.inReadyQueue(tubeName, "{hello:world}"));
+        assertTrue(scheduledJobSystem.inRunningQueue(tubeName, "{hello:world}"));
+
+        // Ack the job.
+        assertTrue(scheduledJobSystem.ackJob(tubeName, "{hello:world}"));
+
+        // Verify the job is in the ready queue even after the ack.
+        assertTrue(scheduledJobSystem.inReadyQueue(tubeName, "{hello:world}"));
+        assertFalse(scheduledJobSystem.inRunningQueue(tubeName, "{hello:world}"));
+
+        JobInfo result2 = scheduledJobSystem.reserveSingle(tubeName, 1000);
+        assertEquals(result2.getJobStr(), "{hello:world}");
+    }
+
+    @Test
+    public void testRemoveExpiredRunningJobs() throws InterruptedException {
+        // Schedule a job
+        scheduledJobSystem.schedule(tubeName, "{hello:world}", 0);
+
+        // Reserve the job.
+        JobInfo result1 = scheduledJobSystem.reserveSingle(tubeName, 500);
+        assertNotNull(result1);
+
+        // Sleep enough time to allow it to expire.
+        Thread.sleep(1000);
```

Comment:
unnecessarily long times.  1 ms reserve ttl w/ Thread.sleep(10ms) is better.  RDBI tests already take too long to run for what the lib is.